File: snow_pkg/src/openalea/Tavg.py
# coding: utf8
from copy import copy

import numpy
from math import *

def model_Tavg(tmin = 0.0,
         tmax = 0.0):
    """
     - Name: Tavg -Version: 1.0, -Time step: 1
     - Description:
                 * Title: Mean temperature  calculation
                 * Author: STICS
                 * Reference: -
                 * Institution: INRA
                 * Abstract: -
     - inputs:
                 * name: tmin
                               ** min : 0.0
                               ** default : 0.0
                               ** max : 500.0
                               ** uri : 
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** inputtype : variable
                               ** unit : degC
                               ** description : current minimum air temperature
                 * name: tmax
                               ** min : 0.0
                               ** default : 0.0
                               ** max : 100.0
                               ** uri : 
                               ** variablecategory : auxiliary
                               ** datatype : DOUBLE
                               ** inputtype : variable
                               ** unit : degC
                               ** description : current maximum air temperature
     - outputs:
                 * name: tavg
                               ** min : 0.0
                               ** variablecategory : auxiliary
                               ** max : 500.0
                               ** uri : 
                               ** datatype : DOUBLE
                               ** unit : degC
                               ** description : mean temperature
    """

    tavg = (tmin + tmax) / 2
    return tavg

File: snow_pkg/src/f90/Tavg.f90
MODULE Tavgmod
    IMPLICIT NONE
CONTAINS

    SUBROUTINE model_Tavg(tmin, &
        tmax, &
        tavg)
        IMPLICIT NONE
        REAL, INTENT(IN) :: tmin
        REAL, INTENT(IN) :: tmax
        REAL, INTENT(OUT) :: tavg
        !- Description:
    !            * Title: Mean temperature  calculation
    !            * Author: STICS
    !            * Reference: -
    !            * Institution: INRA
    !            * Abstract: -
        !- inputs:
    !            * name: tmin
    !                          ** description : current minimum air temperature
    !                          ** inputtype : variable
    !                          ** variablecategory : auxiliary
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 0.0
    !                          ** max : 500.0
    !                          ** unit : degC
    !                          ** uri : 
    !            * name: tmax
    !                          ** description : current maximum air temperature
    !                          ** inputtype : variable
    !                          ** variablecategory : auxiliary
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 0.0
    !                          ** max : 100.0
    !                          ** unit : degC
    !                          ** uri : 
        !- outputs:
    !            * name: tavg
    !                          ** description : mean temperature
    !                          ** variablecategory : auxiliary
    !                          ** datatype : DOUBLE
    !                          ** min : 0.0
    !                          ** max : 500.0
    !                          ** unit : degC
    !                          ** uri : 
        tavg = (tmin + tmax) / 2
    END SUBROUTINE model_Tavg

END MODULE
